src/libraries/System.Private.Xml/src/System/Xml/XmlConvert.cs

```diff
@@ -963,11 +963,17 @@ public static ulong ToUInt64(string s)
 
         public static float ToSingle(string s)
         {
-            s = TrimString(s);
-            if (s == "-INF") return float.NegativeInfinity;
-            if (s == "INF") return float.PositiveInfinity;
-            float f = float.Parse(s, NumberStyles.AllowLeadingSign | NumberStyles.AllowDecimalPoint | NumberStyles.AllowExponent, NumberFormatInfo.InvariantInfo);
-            if (f == 0 && s[0] == '-')
+            ReadOnlySpan<char> roS = s.AsSpan().Trim(WhitespaceChars);
+            switch (roS)
+            {
+                case "-INF":
+                    return float.NegativeInfinity;
+                case "INF":
+                    return float.PositiveInfinity;
+            }
+
+            float f = float.Parse(roS, NumberStyles.AllowLeadingSign | NumberStyles.AllowDecimalPoint | NumberStyles.AllowExponent, NumberFormatInfo.InvariantInfo);
```

Comment:
Nit: I think it would look nicer if this code lived inside a `default` branch in the `switch` statement above.